app/grandchallenge/algorithms/views.py

```diff
@@ -342,41 +393,131 @@ def get_success_url(self):
-            total_jobs = user_credit.credits - jobs["total"]
-        else:
-            total_jobs = user_credit.credits
+    def form_valid(self, form):
+        def create_upload(image_files):
+            raw_files = []
+            upload_session = RawImageUploadSession.objects.create()
+            for image_file in image_files:
+                raw_files.append(
+                    RawImageFile(
+                        upload_session=upload_session,
+                        filename=image_file.name,
+                        staged_file_id=image_file.uuid,
+                    )
+                )
+            RawImageFile.objects.bulk_create(list(raw_files))
+            upload_session.save()
+            return upload_session.pk
 
-        return {
-            "remaining_jobs": int(total_jobs / max(credits_per_job, 1)),
-            "next_job_at": next_job_at,
-            "user_credits": total_jobs,
-        }
+        job = Job.objects.create(
+            creator=self.request.user,
+            algorithm_image=self.algorithm.latest_ready_image,
+        )
+
+        if self.algorithm.editors_group:
+            job.viewer_groups.add(*[self.algorithm.editors_group])
+
+        upload_pks = {}
+        for slug, value in form.cleaned_data.items():
+            ci = self.algorithm.inputs.get(slug=slug)
+            if ci.kind in InterfaceKind.interface_type_image():
+                # create civ without image, image will be added when import completes
+                civ = ComponentInterfaceValue.objects.create(interface=ci)
+                civ.save()
+                job.inputs.add(civ)
+                upload_pks[civ.pk] = create_upload(value)
+            elif ci.kind in InterfaceKind.interface_type_file():
+                # should be a single file
+                civ = ComponentInterfaceValue.objects.create(interface=ci)
+                civ.file = File(
+                    value[0].open(), name=str(Path(ci.output_path).name)
+                )
+                civ.full_clean()
+                civ.save()
+                job.inputs.add(civ)
+            else:
+                civ = ComponentInterfaceValue.objects.create(
+                    interface=ci, value=value
+                )
+                job.inputs.add(civ)
+
+        run_job = run_algorithm_job_for_inputs.signature(
+            kwargs={"job_pk": job.pk, "upload_pks": upload_pks},
+            immutable=True,
+        )
+
+        run_job.apply_async()
+
+        return HttpResponseRedirect(
+            reverse(
+                "algorithms:algorithm-experiment",
+                kwargs={"slug": self.kwargs["slug"], "pk": job.pk},
+            )
+        )
+
+
+class AlgorithmExperiment(
+    LoginRequiredMixin, ObjectPermissionRequiredMixin, TemplateView
+):
+    template_name = "algorithms/experiment.html"
+    permission_required = (
+        f"{Algorithm._meta.app_label}.execute_{Algorithm._meta.model_name}"
+    )
+    raise_exception = True
```

Comment:
The permissions don't seem correct here, shouldn't this be an algorithm job detail view as the job has already been created?